src/Microsoft.Health.Fhir.CosmosDb/Features/Metrics/CosmosMetricProcessor.cs

```diff
@@ -0,0 +1,119 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using AngleSharp.Common;
+using EnsureThat;
+using MediatR;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Health.CosmosDb.Features.Storage;
+using Microsoft.Health.Fhir.Core.Features.Context;
+
+namespace Microsoft.Health.Fhir.CosmosDb.Features.Metrics
+{
+    public class CosmosMetricProcessor : ICosmosMetricProcessor
```

Comment:
Perhaps chagne this name to `CosmosResponseProcessor` or `CosmosDbResponseProcessor` (we should really standardize the naming but doesn't need to be in this one)? It seems like it's doing more than processsing metrics (e.g., it's setting the session token)